internal/hardware/validator.go

```diff
@@ -576,3 +575,41 @@ func compileDiskReasonTemplate(template string, wildcards ...interface{}) *regex
 	}
 	return tmp
 }
+
+// preciseHumanizeBytes formats a byte size using SI units (KB, MB, GB, TB).
+// It displays one decimal place for non-integer values and rounds up to avoid
+// under-reporting the capacity in user-facing messages.
+func preciseHumanizeBytes(bytes uint64) string {
```

Comment:
BTW, go humanize has the function to add decimals: https://github.com/dustin/go-humanize/blob/master/bytes.go#L112-L115